R/workhorse.R

```diff
@@ -44,21 +42,25 @@ set_query <- function(x){
   x
 }
 
-set_body <- function(x){
+set_body <- function(x) {
   if (length(x$params) == 0L) return(x)
 
   x$body <- x$params
   x
 }
 
+
 ## not setting headers yet
 
 set_url <- function(x){
 
-  if (grepl("^http", x$endpoint)){
-    x$url <- x$endpoint
+  if (!is.null(x$endpoint)) {
+    x$url <- file.path(x$api_url,
+                       x$path,
+                       x$endpoint)
   } else{
-    x$url <- file.path(x$api_url, sub(".", "", x$endpoint))
+    x$url <- file.path(x$api_url,
+                       x$path)
```

Comment:
I think you should be using `httr::modify_url()` for this, not `file.path()`.

More like so:
https://github.com/tidyverse/googlesheets2/blob/8caf705a92b7c0638783ce478eb0449a3a92c1de/R/gs_build_request.R#L84-L102

Let me know if I'm missing some big difference between these two APIs.